patternfly/patternfly-elements
# feat: add button variant to pfe-autocomplete

## pfe-autocomplete

**tldr;**
The Red Hat Ecosystem Catalog (RHEC) wants to leverage pfe-autocomplete while keeping the same look and feel of its existing search bar design, which includes the search button to the right of the input field. Adding that variant to the existing pfe-autocomplete component.

### Preview


Link(s) to demo page(s) where this element can be viewed:
- [Demo](https://deploy-preview-1108--happy-galileo-ea79c4.netlify.app/elements/pfe-autocomplete/demo/) 


### What has changed and why

Link to discovery documentation (should contain all the planning and requirements for this work):
   - [Discovery](https://docs.google.com/document/d/1mXOmxgb04jqbDuRwGLTFewyYFxlFDwRmpu4qkZpb2N0/edit#)

### Testing instructions


Go the demo link above and run through the following base scenarios on the textual button example: 

1. Verify that you only see the textual button, not the icon button
2. Verify that the button is disabled until you start typing in the input field
3. Verify that clicking on an item, selects that item, and closes the drop list
4. Verify that that clicking on the clear icon clears the input and disables the button
5. Attach an event listener to pfe-autocomplete to verify that `search-event` is being triggered on click and capturing the search value. 

`document.querySelectorAll('.static')[1].addEventListener('pfe-autocomplete:search-event', (e) => console.log(e.detail.searchValue))`

#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist

Check off items as they are completed.  Feel free to delete items if they are not applicable.



### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
